OSX build fails

https://github.com/datalad/git-annex/runs/4465340747?check_suite_focus=true

```
[24 of 24] Compiling Main             ( Build/Standalone.hs, Build/Standalone.o )
Linking Build/Standalone ...
./Build/Standalone tmp/build-dmg/git-annex.app/Contents/MacOS/bundle tmp/build-dmg/git-annex.app
cp: /usr/lib/libiconv.2.dylib: No such file or directory
installing /usr/local/opt/pcre2/lib/libpcre2-8.0.dylib as A
Standalone: library install failed
CallStack (from HasCallStack):
  error, called at ./Build/OSXMkLibs.hs:73:41 in main:Build.OSXMkLibs
installing /usr/lib/libiconv.2.dylib as B
make: *** [osxapp] Error 1
```
attn @joeyh @jwodder (since I don't know if that is env issue or annex code issue)

--- a/.github/workflows/build-macos.yaml
+++ b/.github/workflows/build-macos.yaml
@@ -177,11 +177,13 @@ jobs:
             See <https://github.com/${{ github.repository }}/actions/runs/${{ github.run_id }}> for more information.
 
   test-annex:
-    runs-on: macos-10.15
+    runs-on: ${{ matrix.os }}
     needs: build-package
     strategy:
       matrix:
         flavor: ["normal", "crippled-tmp"]
+        os: [macos-10.15]
+        include: [{"flavor": "normal", "os": "macos-latest"}]
       fail-fast: false
     steps:
       - name: Checkout this repository
@@ -193,7 +195,7 @@ jobs:
           .github/workflows/tools/set-pr-status \
             "${{ github.event.inputs.pr }}" \
             macOS \
-            "test-annex (${{ matrix.flavor }})" \
+            "test-annex (${{ matrix.flavor }}, ${{ matrix.os }})" \
             pending
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
@@ -270,7 +272,7 @@ jobs:
           .github/workflows/tools/set-pr-status \
             "${{ github.event.inputs.pr }}" \
             macOS \
-            "test-annex (${{ matrix.flavor }})" \
+            "test-annex (${{ matrix.flavor }}, ${{ matrix.os }})" \
             "${{ job.status }}"
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
@@ -287,81 +289,7 @@ jobs:
           to: ${{ secrets.NOTIFY_RECIPIENT }}
           subject: '[${{ github.repository }}] Scheduled tests of macOS build failed!'
           body: |
-            The tests for a scheduled build of git-annex for macOS (flavor: ${{ matrix.flavor }}) failed!
-
-            See <https://github.com/${{ github.repository }}/actions/runs/${{ github.run_id }}> for more information.
-
-  test-annex-next-os:
-    runs-on: macos-latest
-    needs: build-package
-    steps:
-      - name: Checkout this repository
-        uses: actions/checkout@v2
-
-      - name: Create pending PR status
-        if: github.event.inputs.pr != ''
-        run: |
-          .github/workflows/tools/set-pr-status \
-            "${{ github.event.inputs.pr }}" \
-            macOS \
-            test-annex-next-os \
-            pending
-        env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-
-      - name: Download git-annex package
-        uses: actions/download-artifact@v2
-        with:
-          name: git-annex-macos-dmg_${{ needs.build-package.outputs.build-version }}
-
-      - name: Install git-annex package
-        run: |
-          hdiutil attach git-annex_*.dmg
-          rsync -a /Volumes/git-annex/git-annex.app /Applications/
-          hdiutil detach /Volumes/git-annex/
-          echo /Applications/git-annex.app/Contents/MacOS >> "$GITHUB_PATH"
-
-      - name: Install coreutils (for timeout)
-        run: brew install coreutils
-
-      - name: Print git-annex version
-        run: git annex version
-
-      - name: Run tests
-        run: |
-          # Do it after we possibly setup HOME
-          git config --global user.email "test@github.land"
-          git config --global user.name "GitHub Almighty"
-
-          cd $HOME
-          export | grep -e crippledfs || :
-
-          timeout 1800 git annex test
-
-      - name: Set final PR status
-        if: always() && github.event.inputs.pr != ''
-        run: |
-          .github/workflows/tools/set-pr-status \
-            "${{ github.event.inputs.pr }}" \
-            macOS \
-            test-annex-next-os \
-            "${{ job.status }}"
-        env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-
-      - name: Send e-mail on failed scheduled run
-        if: "failure() && github.event_name == 'schedule'"
-        uses: dawidd6/action-send-mail@v2
-        with:
-          server_address: ${{ secrets.NOTIFY_SMTP_HOST }}
-          server_port: ${{ secrets.NOTIFY_SMTP_PORT }}
-          username: ${{ secrets.NOTIFY_SMTP_USERNAME }}
-          password: ${{ secrets.NOTIFY_SMTP_PASSWORD }}
-          from: GitHub Actions Notifications
-          to: ${{ secrets.NOTIFY_RECIPIENT }}
-          subject: '[${{ github.repository }}] Scheduled tests of macOS build failed!'
-          body: |
-            The tests for a scheduled build of git-annex for macOS failed!
+            The tests for a scheduled build of git-annex for macOS (flavor: ${{ matrix.flavor }}, OS: ${{ matrix.os }}) failed!
 
             See <https://github.com/${{ github.repository }}/actions/runs/${{ github.run_id }}> for more information.
 

--- a/.github/workflows/build-macos.yaml
+++ b/.github/workflows/build-macos.yaml
@@ -24,7 +24,7 @@ env:
 
 jobs:
   build-package:
-    runs-on: macos-latest
+    runs-on: macos-10.15
     outputs:
       build-version: ${{ steps.build-version.outputs.version }}
     steps:
@@ -177,7 +177,7 @@ jobs:
             See <https://github.com/${{ github.repository }}/actions/runs/${{ github.run_id }}> for more information.
 
   test-annex:
-    runs-on: macos-latest
+    runs-on: macos-10.15
     needs: build-package
     strategy:
       matrix:
@@ -292,7 +292,7 @@ jobs:
             See <https://github.com/${{ github.repository }}/actions/runs/${{ github.run_id }}> for more information.
 
   test-annex-more:
-    runs-on: macos-latest
+    runs-on: macos-10.15
     needs: build-package
     steps:
       - name: Checkout this repository
@@ -345,7 +345,7 @@ jobs:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
 
   test-datalad:
-    runs-on: macos-latest
+    runs-on: macos-10.15
     needs: build-package
     strategy:
       matrix:

--- a/.github/workflows/build-macos.yaml
+++ b/.github/workflows/build-macos.yaml
@@ -291,6 +291,80 @@ jobs:
 
             See <https://github.com/${{ github.repository }}/actions/runs/${{ github.run_id }}> for more information.
 
+  test-annex-next-os:
+    runs-on: macos-latest
+    needs: build-package
+    steps:
+      - name: Checkout this repository
+        uses: actions/checkout@v2
+
+      - name: Create pending PR status
+        if: github.event.inputs.pr != ''
+        run: |
+          .github/workflows/tools/set-pr-status \
+            "${{ github.event.inputs.pr }}" \
+            macOS \
+            test-annex-next-os \
+            pending
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Download git-annex package
+        uses: actions/download-artifact@v2
+        with:
+          name: git-annex-macos-dmg_${{ needs.build-package.outputs.build-version }}
+
+      - name: Install git-annex package
+        run: |
+          hdiutil attach git-annex_*.dmg
+          rsync -a /Volumes/git-annex/git-annex.app /Applications/
+          hdiutil detach /Volumes/git-annex/
+          echo /Applications/git-annex.app/Contents/MacOS >> "$GITHUB_PATH"
+
+      - name: Install coreutils (for timeout)
+        run: brew install coreutils
+
+      - name: Print git-annex version
+        run: git annex version
+
+      - name: Run tests
+        run: |
+          # Do it after we possibly setup HOME
+          git config --global user.email "test@github.land"
+          git config --global user.name "GitHub Almighty"
+
+          cd $HOME
+          export | grep -e crippledfs || :
+
+          timeout 1800 git annex test
+
+      - name: Set final PR status
+        if: always() && github.event.inputs.pr != ''
+        run: |
+          .github/workflows/tools/set-pr-status \
+            "${{ github.event.inputs.pr }}" \
+            macOS \
+            test-annex-next-os \
+            "${{ job.status }}"
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Send e-mail on failed scheduled run
+        if: "failure() && github.event_name == 'schedule'"
+        uses: dawidd6/action-send-mail@v2
+        with:
+          server_address: ${{ secrets.NOTIFY_SMTP_HOST }}
+          server_port: ${{ secrets.NOTIFY_SMTP_PORT }}
+          username: ${{ secrets.NOTIFY_SMTP_USERNAME }}
+          password: ${{ secrets.NOTIFY_SMTP_PASSWORD }}
+          from: GitHub Actions Notifications
+          to: ${{ secrets.NOTIFY_RECIPIENT }}
+          subject: '[${{ github.repository }}] Scheduled tests of macOS build failed!'
+          body: |
+            The tests for a scheduled build of git-annex for macOS failed!
+
+            See <https://github.com/${{ github.repository }}/actions/runs/${{ github.run_id }}> for more information.
+
   test-annex-more:
     runs-on: macos-10.15
     needs: build-package

--- a/.github/workflows/build-ubuntu.yaml
+++ b/.github/workflows/build-ubuntu.yaml
@@ -183,11 +183,12 @@ jobs:
             See <https://github.com/${{ github.repository }}/actions/runs/${{ github.run_id }}> for more information.
 
   test-annex:
-    runs-on: ubuntu-latest
+    runs-on: ${{ matrix.os }}
     needs: build-package
     strategy:
       matrix:
         flavor: ["normal", "crippled-tmp", "crippled-home", "nfs-home"]
+        os: [ubuntu-latest]
       fail-fast: false
     steps:
       - name: Checkout this repository
@@ -199,7 +200,7 @@ jobs:
           .github/workflows/tools/set-pr-status \
             "${{ github.event.inputs.pr }}" \
             Ubuntu \
-            "test-annex (${{ matrix.flavor }})" \
+            "test-annex (${{ matrix.flavor }}, ${{ matrix.os }})" \
             pending
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
@@ -270,7 +271,7 @@ jobs:
           .github/workflows/tools/set-pr-status \
             "${{ github.event.inputs.pr }}" \
             Ubuntu \
-            "test-annex (${{ matrix.flavor }})" \
+            "test-annex (${{ matrix.flavor }}, ${{ matrix.os }})" \
             "${{ job.status }}"
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
@@ -287,7 +288,7 @@ jobs:
           to: ${{ secrets.NOTIFY_RECIPIENT }}
           subject: '[${{ github.repository }}] Scheduled tests of Ubuntu build failed!'
           body: |
-            The tests for a scheduled build of git-annex for Ubuntu (flavor: ${{ matrix.flavor }}) failed!
+            The tests for a scheduled build of git-annex for Ubuntu (flavor: ${{ matrix.flavor }}, OS: ${{ matrix.os }}) failed!
 
             See <https://github.com/${{ github.repository }}/actions/runs/${{ github.run_id }}> for more information.
 

--- a/.github/workflows/template/build-{{ostype}}.yaml.j2
+++ b/.github/workflows/template/build-{{ostype}}.yaml.j2
@@ -454,6 +454,81 @@ jobs:
 
             See <https://github.com/${{ github.repository }}/actions/runs/${{ github.run_id }}> for more information.
 
+{% if ostype == "macos" %}
+  test-annex-next-os:
+    runs-on: macos-latest
+    needs: build-package
+    steps:
+      - name: Checkout this repository
+        uses: actions/checkout@v2
+
+      - name: Create pending PR status
+        if: github.event.inputs.pr != ''
+        run: |
+          .github/workflows/tools/set-pr-status \
+            "${{ github.event.inputs.pr }}" \
+            {{osname}} \
+            test-annex-next-os \
+            pending
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Download git-annex package
+        uses: actions/download-artifact@v2
+        with:
+          name: {{artifact_basename}}_${{ needs.build-package.outputs.build-version }}
+
+      - name: Install git-annex package
+        run: |
+          {% for step in artifact_install_steps %}
+          {{step}}
+          {% endfor %}
+
+      - name: Install coreutils (for timeout)
+        run: brew install coreutils
+
+      - name: Print git-annex version
+        run: git annex version
+
+      - name: Run tests
+        run: |
+          # Do it after we possibly setup HOME
+          git config --global user.email "test@github.land"
+          git config --global user.name "GitHub Almighty"
+
+          cd $HOME
+          export | grep -e crippledfs || :
+
+          timeout 1800 git annex test
+
+      - name: Set final PR status
+        if: always() && github.event.inputs.pr != ''
+        run: |
+          .github/workflows/tools/set-pr-status \
+            "${{ github.event.inputs.pr }}" \
+            {{osname}} \
+            test-annex-next-os \
+            "${{ job.status }}"
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Send e-mail on failed scheduled run
+        if: "failure() && github.event_name == 'schedule'"
+        uses: dawidd6/action-send-mail@v2
+        with:
+          server_address: ${{ secrets.NOTIFY_SMTP_HOST }}
+          server_port: ${{ secrets.NOTIFY_SMTP_PORT }}
+          username: ${{ secrets.NOTIFY_SMTP_USERNAME }}
+          password: ${{ secrets.NOTIFY_SMTP_PASSWORD }}
+          from: GitHub Actions Notifications
+          to: ${{ secrets.NOTIFY_RECIPIENT }}
+          subject: '[${{ github.repository }}] Scheduled tests of {{osname}} build failed!'
+          body: |
+            The tests for a scheduled build of git-annex for {{osname}} failed!
+
+            See <https://github.com/${{ github.repository }}/actions/runs/${{ github.run_id }}> for more information.
+
+{% endif %}
 {% if ostype == "ubuntu" or ostype == "macos" %}
   test-annex-more:
     runs-on: {{runs_on}}

--- a/.github/workflows/template/build-{{ostype}}.yaml.j2
+++ b/.github/workflows/template/build-{{ostype}}.yaml.j2
@@ -307,13 +307,20 @@ jobs:
             See <https://github.com/${{ github.repository }}/actions/runs/${{ github.run_id }}> for more information.
 
   test-annex:
-    runs-on: {{runs_on}}
-    needs: build-package
   {% if test_annex_flavors %}
+    runs-on: ${{ matrix.os }}
+    needs: build-package
     strategy:
       matrix:
         flavor: {{test_annex_flavors|tojson}}
+        os: [{{runs_on}}]
+      {% if test_annex_include %}
+        include: {{test_annex_include|tojson}}
+      {% endif %}
       fail-fast: false
+  {% else %}
+    runs-on: {{runs_on}}
+    needs: build-package
   {% endif %}
     steps:
       - name: Checkout this repository
@@ -326,7 +333,7 @@ jobs:
             "${{ github.event.inputs.pr }}" \
             {{osname}} \
             {% if test_annex_flavors %}
-            "test-annex (${{ matrix.flavor }})" \
+            "test-annex (${{ matrix.flavor }}, ${{ matrix.os }})" \
             {% else %}
             test-annex \
             {% endif %}
@@ -426,7 +433,7 @@ jobs:
             "${{ github.event.inputs.pr }}" \
             {{osname}} \
             {% if test_annex_flavors %}
-            "test-annex (${{ matrix.flavor }})" \
+            "test-annex (${{ matrix.flavor }}, ${{ matrix.os }})" \
             {% else %}
             test-annex \
             {% endif %}
@@ -447,88 +454,13 @@ jobs:
           subject: '[${{ github.repository }}] Scheduled tests of {{osname}} build failed!'
           body: |
           {% if test_annex_flavors %}
-            The tests for a scheduled build of git-annex for {{osname}} (flavor: ${{ matrix.flavor }}) failed!
+            The tests for a scheduled build of git-annex for {{osname}} (flavor: ${{ matrix.flavor }}, OS: ${{ matrix.os }}) failed!
           {% else %}
             The tests for a scheduled build of git-annex for {{osname}} failed!
           {% endif %}
 
             See <https://github.com/${{ github.repository }}/actions/runs/${{ github.run_id }}> for more information.
 
-{% if ostype == "macos" %}
-  test-annex-next-os:
-    runs-on: macos-latest
-    needs: build-package
-    steps:
-      - name: Checkout this repository
-        uses: actions/checkout@v2
-
-      - name: Create pending PR status
-        if: github.event.inputs.pr != ''
-        run: |
-          .github/workflows/tools/set-pr-status \
-            "${{ github.event.inputs.pr }}" \
-            {{osname}} \
-            test-annex-next-os \
-            pending
-        env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-
-      - name: Download git-annex package
-        uses: actions/download-artifact@v2
-        with:
-          name: {{artifact_basename}}_${{ needs.build-package.outputs.build-version }}
-
-      - name: Install git-annex package
-        run: |
-          {% for step in artifact_install_steps %}
-          {{step}}
-          {% endfor %}
-
-      - name: Install coreutils (for timeout)
-        run: brew install coreutils
-
-      - name: Print git-annex version
-        run: git annex version
-
-      - name: Run tests
-        run: |
-          # Do it after we possibly setup HOME
-          git config --global user.email "test@github.land"
-          git config --global user.name "GitHub Almighty"
-
-          cd $HOME
-          export | grep -e crippledfs || :
-
-          timeout 1800 git annex test
-
-      - name: Set final PR status
-        if: always() && github.event.inputs.pr != ''
-        run: |
-          .github/workflows/tools/set-pr-status \
-            "${{ github.event.inputs.pr }}" \
-            {{osname}} \
-            test-annex-next-os \
-            "${{ job.status }}"
-        env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-
-      - name: Send e-mail on failed scheduled run
-        if: "failure() && github.event_name == 'schedule'"
-        uses: dawidd6/action-send-mail@v2
-        with:
-          server_address: ${{ secrets.NOTIFY_SMTP_HOST }}
-          server_port: ${{ secrets.NOTIFY_SMTP_PORT }}
-          username: ${{ secrets.NOTIFY_SMTP_USERNAME }}
-          password: ${{ secrets.NOTIFY_SMTP_PASSWORD }}
-          from: GitHub Actions Notifications
-          to: ${{ secrets.NOTIFY_RECIPIENT }}
-          subject: '[${{ github.repository }}] Scheduled tests of {{osname}} build failed!'
-          body: |
-            The tests for a scheduled build of git-annex for {{osname}} failed!
-
-            See <https://github.com/${{ github.repository }}/actions/runs/${{ github.run_id }}> for more information.
-
-{% endif %}
 {% if ostype == "ubuntu" or ostype == "macos" %}
   test-annex-more:
     runs-on: {{runs_on}}

--- a/.github/workflows/template/specs.yml
+++ b/.github/workflows/template/specs.yml
@@ -24,6 +24,9 @@
     # TODO: Add "crippled-home" back in once
     # <https://git-annex.branchable.com/bugs/gets_stuck_in_Remote_Tests___40____63____41___while_running_on_OSX_with_HOME_on_crippled_FS/?updated>
     # is fixed
+  test_annex_include:
+    - flavor: normal
+      os: macos-latest
   artifact_basename: git-annex-macos-dmg
   artifact_install_steps:
     - hdiutil attach git-annex_*.dmg

--- a/.github/workflows/template/specs.yml
+++ b/.github/workflows/template/specs.yml
@@ -17,7 +17,7 @@
 - ostype: macos
   osname: macOS
   cron_hour: '01'
-  runs_on: macos-latest
+  runs_on: macos-10.15
   env:
     LANG: C
   test_annex_flavors: [normal, crippled-tmp]

